### ISSUE TITLE
Add support for JavaDoc tags @apiNote, @implSpec and @implNote

### ISSUE THREAD

**Issue by uhafner** [google/error-prone#1796]
### Description of the problem / feature request:

The rule `InvalidTag` reports false positives for JavaDoc tag `@implNote`. 

Error message:
```
 [InvalidTag] Tag name `implNote` is unknown. If this is a commonly-used custom tag, please click 'not useful' and file a bug.
```

### Feature requests: what underlying problem are you trying to solve with this feature?

Please support these tags: https://blog.codefx.org/java/new-javadoc-tags/

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
    /**
     * Detects variations of {@code rootProject.setName($string)}, as well as the Groovy setter shorthand
     * {@code rootProject.name = $string}.
     *
     * @implNote This expression does not guarantee matching left/right quotes or parentheses. For example, it will
     * match {@code 'Name"}, even though that is not valid Groovy or Kotlin syntax. Similarly, in allowing for Groovy's
     * parentheses-optional syntax for {@code setName $string}, this pattern makes both open and close parentheses
     * optional, rather than requiring neither or both, so it will parse an open parenthesis without a matching close,
     * and vice versa.
     */
     public class Tags {}
```

### What version of Error Prone are you using?

2.4.0

###  Have you found anything relevant by searching the web?

No

**Comment by graememorgan** [comment]
I think this does work at HEAD, where the check is called `InvalidBlockTag`. That should be in 2.4.0.

**Comment by uhafner** [comment]
In my build with 2.4.0 I still get the error message with `InvalidTag`: https://ci.jenkins.io/job/Plugins/job/analysis-model/job/master/1085/consoleFull 

So should I simply deactivate the rule `InvalidTag` as it has been deprecated?

**Comment by cushon** [comment]
Are you sure you're using 2.4.0? I think that check was deleted before that release: https://github.com/google/error-prone/commit/57915048ab23de6918857cb201fff372567812ec

**Comment by uhafner** [comment]
Hmm, interesting. Seems that then my maven configuration is broken indeed. I must admit that from the readme I did not find out how to configure it correctly with maven and Java 8. Seems that an old version of error prone is picked up.

**Comment by Stephan202** [comment]
@uhafner for a Maven configuration with Error Prone that is both JDK 8 and JDK 11 compatible you could have a look at [this `pom.xml`](https://github.com/PicnicSupermarket/oss-parent/blob/master/pom.xml) The [`jdk8` profile](https://github.com/PicnicSupermarket/oss-parent/blob/5df07a52ef6ee20e44850e09de58e2245b8c9935/pom.xml#L832-L864) shows what's necessary for Java 8. (Ignore the `verification.skip` reference.)

**Comment by uhafner** [comment]
Thanks @Stephan202, I actually found the bug in my configurations. I still used `plexus-compiler-javac-errorprone` which includes an old error prone version. Now everything is working correctly!
